### PR TITLE
[Test] Add ssat tests for tensordec-imagesegment 

### PR DESCRIPTION
**Pull request by anyj0527**
Add a fail case, a simple-run case, and simple golden compare test with videotestsrc.

See also: #1813 

Signed-off-by: Yongjoo Ahn <yongjoo1.ahn@samsung.com>

### CODE REVIEW

File: tests/nnstreamer_decoder_image_segment/runTest.sh [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+
+# THIS WON'T FAIL, BUT NOT MUCH MEANINGFUL.
+gstTest "--gst-plugin-path=${PATH_TO_PLUGIN} videotestsrc num_buffers=4 ! videoconvert ! videoscale ! video/x-raw,format=RGB,width=257,height=257 ! tee name=t t. ! queue ! mix. t. ! queue ! tensor_converter ! tensor_transform mode=arithmetic option=typecast:float32,div:255.0 ! tensor_filter framework=tensorflow-lite model=${PATH_TO_MODEL} ! tensor_decoder mode=image_segment option1=tflite-deeplab ! mix. videomixer name=mix sink_0::alpha=0.7 sink_1::alpha=0.6 ! videoconvert ! fakesink" 0_p 0 0
+
+gstTest "--gst-plugin-path=${PATH_TO_PLUGIN} videotestsrc num-buffers=1 ! videoconvert ! videoscale ! video/x-raw,format=RGB,width=257,height=257 ! tee name=t t. ! queue ! mix. t. ! queue ! tensor_converter ! tensor_transform mode=arithmetic option=typecast:float32,div:255.0 ! tensor_filter framework=tensorflow-lite model=${PATH_TO_MODEL} ! tensor_decoder mode=image_segment option1=tflite-deeplab ! mix. videomixer name=mix sink_0::alpha=0.7 sink_1::alpha=0.6 ! filesink location=test_output.0" 1 0 0 $PERFORMANCE
+
+callCompareTest test_golden.0 test_output.0 1 "test with videotestsrc" 0
```
Comment (dongju-chae) [inline tests/nnstreamer_decoder_image_segment/runTest.sh]: How about moving this golden data, `test_golden.0`, to the data folder, `tests/test_models/models/data`

Comment (anyj0527) [reply tests/nnstreamer_decoder_image_segment/runTest.sh]: I found that every other SSAT golden data is located in the directory which its runTest.sh file exists. Do I need to move all those files to the new path?

Comment (dongju-chae) [reply tests/nnstreamer_decoder_image_segment/runTest.sh]: Then, it's fine :)

File: tests/nnstreamer_decoder_image_segment/runTest.sh [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+fi
+
+# This is compatible with SSAT (https://github.com/myungjoo/SSAT)
+testInit $1
+
+# NNStreamer and plugins path for test
+PATH_TO_PLUGIN="../../build"
+
+if [[ -d $PATH_TO_PLUGIN ]]; then
+    ini_path="${PATH_TO_PLUGIN}/ext/nnstreamer/tensor_filter"
```
Comment (wooksong) [inline tests/nnstreamer_decoder_image_segment/runTest.sh]: FYI, "${PATH_TO_PLUGIN}/ext/nnstreamer/tensor_filter" is not ini_path.

**Comment by wooksong** [comment]
> Instead of including tests/nnstreamer_decoder_image_segment/test_golden.0 in the git repo, it is better to be generated by the tool when it is needed.

Resolved.

**Pull request merged**